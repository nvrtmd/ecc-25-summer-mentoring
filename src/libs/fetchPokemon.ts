import { PokemonSchema } from '@/schemas'

export async function fetchPokemonByName(name: string) {
  const res = await fetch(`https://pokeapi.co/api/v2/pokemon/${name}`)
  if (!res.ok) throw new Error('Pokemon not found')

  const json = await res.json()
  return PokemonSchema.parse(json)
}
