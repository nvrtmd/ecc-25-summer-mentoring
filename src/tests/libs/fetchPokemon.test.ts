import { fetchPokemonByName } from '@/libs/fetchPokemon'
import { describe, expect, test } from 'vitest'

describe('fetchPokemonByName', () => {
  test('fetchPokemonByName - pikachu', async () => {
    const pokemon = await fetchPokemonByName('pikachu')
    expect(pokemon).toMatchSnapshot()
  })

  test('fetchPokemonByName - pichu', async () => {
    const pokemon = await fetchPokemonByName('pichu')
    expect(pokemon.name).toBe('pichu')
    expect(pokemon.height).toBe(3)
    expect(pokemon.weight).toBe(20)
    expect(pokemon.abilities).toHaveLength(2)
    expect(pokemon.stats).toHaveLength(6)
  })
})
