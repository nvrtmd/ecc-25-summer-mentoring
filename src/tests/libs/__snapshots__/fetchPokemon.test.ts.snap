// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fetchPokemonByName > fetchPokemonByName - pikachu 1`] = `
{
  "abilities": [
    {
      "ability": {
        "name": "static",
        "url": "https://pokeapi.co/api/v2/ability/9/",
      },
      "is_hidden": false,
      "slot": 1,
    },
    {
      "ability": {
        "name": "lightning-rod",
        "url": "https://pokeapi.co/api/v2/ability/31/",
      },
      "is_hidden": true,
      "slot": 3,
    },
  ],
  "height": 4,
  "id": 25,
  "name": "pikachu",
  "stats": [
    {
      "base_stat": 35,
      "effort": 0,
      "stat": {
        "name": "hp",
        "url": "https://pokeapi.co/api/v2/stat/1/",
      },
    },
    {
      "base_stat": 55,
      "effort": 0,
      "stat": {
        "name": "attack",
        "url": "https://pokeapi.co/api/v2/stat/2/",
      },
    },
    {
      "base_stat": 40,
      "effort": 0,
      "stat": {
        "name": "defense",
        "url": "https://pokeapi.co/api/v2/stat/3/",
      },
    },
    {
      "base_stat": 50,
      "effort": 0,
      "stat": {
        "name": "special-attack",
        "url": "https://pokeapi.co/api/v2/stat/4/",
      },
    },
    {
      "base_stat": 50,
      "effort": 0,
      "stat": {
        "name": "special-defense",
        "url": "https://pokeapi.co/api/v2/stat/5/",
      },
    },
    {
      "base_stat": 90,
      "effort": 2,
      "stat": {
        "name": "speed",
        "url": "https://pokeapi.co/api/v2/stat/6/",
      },
    },
  ],
  "weight": 60,
}
`;
